### PR TITLE
fix(asurascans): fix chapter names and page list parsing

### PR DESCRIPTION
**Pull request by AlokSwaminathan**
Checklist:
- [x] Updated source's version for individual source changes
- [x] Updated all sources' versions for template changes
- [x] Set appropriate `nsfw` value
- [x] Did not change `id` even if a source's name or language were changed
- [x] Tested the modifications by running it on the simulator or a test device

### CODE REVIEW

File: src/rust/en.asurascans/src/lib.rs [hunk 1]
```diff
@@ -299,31 +299,41 @@ fn get_chapter_list(manga_id: String) -> Result<Vec<Chapter>> {
...
-			index,
-			url,
-			..Default::default()
-		});
+  let mut text_slice = html_text.as_str();
+  loop {
+    let chap = text_slice.find("https://gg.asuracomic.net/storage/media/");
+    if let Some(chap) = chap {
+      text_slice = &text_slice[chap..];
+      let end = text_slice.find(".webp").unwrap();
```
Comment (Skittyblock) [inline src/rust/en.asurascans/src/lib.rs:311]: don't use `unwrap`, if it doesn't exist either continue or return early with an error

**Review by Skittyblock — CHANGES_REQUESTED** [review]
make sure clippy passes; i think it will give you some error about the indentation in lib.rs

edit: i guess clippy passed in the workflow? make sure the file is properly indented though

**Comment by AlokSwaminathan** [comment]
I had some space based tabbing on accident, I think that's fixed now.

**Review by Skittyblock — APPROVED** [review]
i guess we can have this as a temporary fix

**Pull request merged**